spring-integration-core/src/main/java/org/springframework/integration/transformer/ObjectToMapTransformer.java

```diff
@@ -52,14 +54,23 @@
  * @author Oleg Zhurakousky
  * @author Artem Bilan
  * @author Gary Russell
+ * @author Vikas Prasad
  * @since 2.0
  */
 public class ObjectToMapTransformer extends AbstractPayloadTransformer<Object, Map<?, ?>> {
 
-	private final JsonObjectMapper<?, ?> jsonObjectMapper = JsonObjectMapperProvider.newInstance();
+	private final JsonObjectMapper<?, ?> jsonObjectMapper;
 
 	private volatile boolean shouldFlattenKeys = true;
 
+	public ObjectToMapTransformer() {
+		this.jsonObjectMapper = JsonObjectMapperProvider.newInstance();
+	}
+
+	public ObjectToMapTransformer(ObjectMapper objectMapper) {
```

Comment:
That's not good. We advertise this class as JSON-engine independent. So, if we let to inject something that really should be `JsonObjectMapper`, but not target Jackson `ObjectMapper`.

Also the `Assert.notNull()` must be here do not allow to inject `null` for potential `NPE`s in the target code.